docs/specs/meta-clients/web3modal/api.md

```diff
@@ -1,142 +1,248 @@
-# API
+# API (v2)
 
-Web3Modal has it's own api available at [api.web3modal.com](https://api.web3modal.com) that is used to fetch wallet data from a WalletConnect maintained [wallet & dapp registry](https://walletconnect.com/explorer).
+> For the deprecated v1 specification, see [here](./api-v1-legacy.md).
 
-All responses are cached for a minimum of 2h.
+## Query Parameters
+
+All API requests now use query parameters instead of custom headers for SDK metadata, making integration simpler and more compatible with common HTTP clients and browser caching mechanisms.
+
+### Mandatory Query Parameters
+
+All API endpoints expect the following mandatory query parameters:
+
+- `st=`: SDK type
+  - Options: `w3m | wcm`
+  - Example: `st=w3m`
+- `sf=`: SDK framework
+  - Options: `html | react | vue`
+  - Example: `sf=react`
+- `sa=`: SDK adapter
+  - Options: `evm-wagmi | evm-ethers | evm-ethers5`
+  - Example: `sa=evm-wagmi`
+- `sv=`: SDK version
+  - Options: `string`
+  - Example: `sv=5.0.0`
+- `projectId=`: The project ID for tracking and analytics
+  - Options: `string`
+  - Example: `projectId=702e2d45...`
 
 ## Headers
 
-All api requests must contain following headers. See [types](./types.md)
+All API endpoints expect the following simple (i.e. non-preflight-dependent) headers:
 
-```ts
-interface Headers {
-  "x-project-id": string;
-  "x-sdk-type": SdkType;
-  "x-sdk-version": SdkVersion;
-}
-```
+- `Accept-Encoding`
+  - Options: `gzip | br`
+  - Example: `Accept-Encoding: gzip, br`
 
-## Methods
+## API Endpoints
 
-### GET `/getWallets`
+### `POST /v2/wallets`
 
-Returns total count and wallet data, based on provided body inputs. See [types](./types.md).
+Fetches wallet data based on following request body provided as a stringified JSON object (`Content-Type: text/plain`).
 
-```ts
-interface Request {
-  headers: Headers;
-}
+#### Request Body
 
-interface UrlQueryParams {
-  page: string; // eg. 1
-  entries: string; // eg. 100
-  search?: string; // eg. MetaMa...
-  include?: string; // eg. id1,id2,id3
-  exclude?: string; // eg. id1,id2,id3
-  chains?: string; // eg. eip155:1,solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp
-  platform?: "ios" | "android";
+```typescript
+{
+  "page": number, // Required - The page number of results (e.g., `1`)
+  "entries": number, // Required - Number of entries per page (e.g., `100`)
+  "search": string | null, // Optional search term (e.g., `MetaMask`)
+  "include": string | null, // Optional comma-separated list of wallet IDs to include
+  "exclude": string | null, // Optional comma-separated list of wallet IDs to exclude
+  "chains": string | null, // Optional comma-separated list of chain identifiers (e.g., `eip155:1`)
```

Comment:
We should also document the behaviour to improve DX when they provide these query params.
Eg what's the logical operator used underneath, what happens if you provide `exclude` and `search` for an excluded `walletId`, etc.